google-cloud-logging/lib/google/cloud/logging/middleware.rb

```diff
@@ -73,39 +73,102 @@ def extract_trace_id env
         end
 
         ##
+        # Construct a monitored resource based on the given type and label if
+        # both are provided. Otherwise, construct a default monitored resource
+        # based on the current environment.
+        #
+        # @example
+        #   # If both type and labels are provided, it returns resource:
+        #   <Google::Cloud::Logging::Resource
+        #     @type=[given type],
+        #     @labels=[given labels] >
+        #
+        #   # Otherwise, if running from GAE, returns resource:
+        #   <Google::Cloud::Logging::Resource
+        #     @type="gae_app",
+        #     @labels={
+        #       module_id: [GAE module name],
+        #       version_id: [GAE module version] }>
+        #
+        #   # If running from GKE, returns resource:
+        #   <Google::Cloud::Logging::Resource
+        #     @type="container",
+        #     @labels={
+        #       cluster_name: [GKE cluster name],
+        #       namespace_id: [GKE namespace_id] }>
+        #
+        #   # If running from GCE, return resource:
+        #   <Google::Cloud::Logging::Resource
+        #     @type="gce_instance",
+        #     @labels={
+        #       instance_id: [GCE VM instance id],
+        #       zone: [GCE vm group zone] }>
+        #
+        #   # Otherwise default to "global" type, which means not
+        #   # associated with GCP:
+        #   <Google::Cloud::Logging::Resource
+        #     @type="global",
+        #     @labels={}>
```

Comment:
Sure, I can split it up into multiple examples. But I had to shorten the wording a little bit to make it consistent and fit under 80 line width.